File: ELEC881_Tabnine/src/No Description Prompt/Python/q27-number-of-digit-one.py
class Solution(object):
    def countDigitOne(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        i = 1
        while i <= n:
            a = n // i
            b = n % i
            count += (a + 8) // 10 * i + (a % 10 == 1) * (b + 1)
            i *= 10
        return count
        
    

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q27-number-of-digit-one.js
/**
 * @param {number} n
 * @return {number}
 */
var countDigitOne = function(n) {
    let count = 0;
    for (let i = 1; i <= n; i *= 10) {
        let div = i * 10;
        let mod = n % div;
        let high = Math.floor(n / div);
        let cur = Math.floor(high / 10) * i + high % 10 * 1 + mod % i + 1;
        count += (high % 10) * i + mod % i + 1;
        if (cur === n) break;
    }
    return count;
};